src/odemis/gui/cont/multi_point_correlation.py

```diff
@@ -0,0 +1,762 @@
+                    self.correlation_target.fm_pois.append(target)
+            if fib_fiducials:
+                fib_fiducials.sort(key=lambda x: x.index.value)
+                self.correlation_target.fib_fiducials = fib_fiducials
+            if fm_fiducials:
+                fm_fiducials.sort(key=lambda x: x.index.value)
+                self.correlation_target.fm_fiducials = fm_fiducials
+
+        acq_conf = conf.get_acqui_conf()
+        save_features(acq_conf.pj_last_path, self._tab_data_model.main.features.value)
+
+    def check_correlation_conditions(self) -> bool:
+        """
+        Minimum 4 FIB and FM fiducials and 1 POI in FM are required to run the correlation.
+        :return: (bool) True if the conditions are met, False otherwise
+        """
+        if self.correlation_target:
+            if ((len(self.correlation_target.fib_fiducials) >= 4 and len(
+                    self.correlation_target.fm_fiducials) >= 4 and len(self.correlation_target.fm_pois) >=1  and len(
+                self._tab_data_model.views.value[0].stream_tree) > 0) and self.correlation_target.fib_stream and
+                    (len(self.correlation_target.fm_fiducials) == len(self.correlation_target.fib_fiducials))):
+                return True
+            else:
+                self.correlation_target.clear()
+                self._tab_data_model.projected_points = []
+                self.correlation_txt.SetLabel("To run correlation, please add \n"
+                                              "minimum 4 FIB-FM fiducial pairs, 1 POI in FM.")
+                self._panel.Layout()
+                # Update the FIB viewport because it shows the output overlays
+                # It is the second viewport out of total two viewports
+                self._viewports[1].canvas.update_drawing()
+                return False
+        else:
+            return False
+
+    @wxlimit_invocation(0.05)
+    def _queue_latest_change(self):
+        """Queue the latest change to process the correlation result."""
+        while not self.change_queue.empty():
+            try:
+                self.change_queue.get_nowait()  # Remove older, outdated requests
+            except queue.Empty:
+                break
+        self.change_queue.put(True)  # Only enqueue the latest change request
+
+    def _process_queue(self):
+        """Worker thread that continuously processes requests from the queue."""
+        while True:
+            task = self.change_queue.get()
+            if task is None:  # Special exit signal
+                break
+            self._process_latest_change()
+
+    def _process_latest_change(self):
+        """Process the latest change in the queue."""
+        self.is_processing = True
+        self._do_correlation()
+        rms = self.correlation_target.correlation_result["output"]["error"]["rms_error"]
+        self.correlation_txt.SetLabel(f"Correlation RMS Deviation : {readable_str(rms, sig=3)}")
```

Comment:
This touches the GUI, and it's run in a separate thread => It's going to crash the GUI! 
As you only have 1 line that changes the GUI, the simplest might be to use a `wx.CallAfter()`, like that:
```
wx.CallAfter(self.correlation_txt.SetLabel,
              f"Correlation RMS Deviation : {readable_str(rms, sig=3)}")
```
(If you had more lines, you'd typically move them to a separate function, protected by `@call_in_wx_main`)